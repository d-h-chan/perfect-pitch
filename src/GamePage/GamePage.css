.circle {
  width: 200px;
  height: 200px;
  border-radius: 50%;
}
.circle .note {
  position: absolute;
  border-radius: 50%;
  left: 50%;
  top: 50%;
  margin-left: -20px;
  margin-top: 20px;
  width: 45px;
  height: 45px;
  font-size: 18px;
  font-family: "Quicksand";
  border: 1px solid lightgray;
  
}

#playNoteButton {
  position: absolute;
  border-radius: 50%;
  left: 50%;
  top: 50%;
  margin-left: -47.5px;
  margin-top: -10px;
  width: 100px;
  height: 100px;
  font-size: 18px;
  font-family: "Quicksand";
  border: 1px solid lightgray;

}

.circle .note:nth-child(4) {
  transform: rotate(0deg) translate(110px);
  background: linear-gradient(to bottom, #9D8FD4 0%, #BA88D6 100%)
}
.circle .note:nth-child(5) {
  transform: rotate(30deg) translate(110px) rotate(-30deg);
  background: linear-gradient(to bottom left, #BA88D6 0%, #E587BB 100%)
}
.circle .note:nth-child(6) {
  transform: rotate(60deg) translate(110px) rotate(-60deg);
  background: linear-gradient(to bottom left, #E587BB 0%, #FE8786 100%)
}
.circle .note:nth-child(7) {
  transform: rotate(90deg) translate(110px) rotate(-90deg);
  background: linear-gradient(to left, #FE8786 0%, #FFBA86 100%)
}
.circle .note:nth-child(8) {
  transform: rotate(120deg) translate(110px) rotate(-120deg);
  background: linear-gradient(to top left, #FFBA86 0%, #FDD68D 100%)
}
.circle .note:nth-child(9) {
  transform: rotate(150deg) translate(110px) rotate(-150deg);
  background: linear-gradient(to top left, #FDD68D 0%, #FBEA82 100%)
}
.circle .note:nth-child(10) {
  transform: rotate(180deg) translate(110px) rotate(-180deg);
  background: linear-gradient(to top, #FBEA82 0%, #FFFB89 100%)
}
.circle .note:nth-child(11) {
  transform: rotate(210deg) translate(110px) rotate(-210deg);
  background: linear-gradient(to top right, #FFFB89 0%, #CCF686 100%)
}
.circle .note:nth-child(12) {
  transform: rotate(240deg) translate(110px) rotate(-240deg);
  background: linear-gradient(to top right, #CCF686 0%, #89E485 100%)
}
.circle .note:nth-child(1) {
  transform: rotate(270deg) translate(110px) rotate(-270deg);
  background: linear-gradient(to right, #89E485 0%, #88CCCF 100%)
}
.circle .note:nth-child(2) {
  transform: rotate(300deg) translate(110px) rotate(-300deg);
  background: linear-gradient(to bottom right, #88CCCF 0%, #8EA3D5 100%)
}
.circle .note:nth-child(3) {
  transform: rotate(330deg) translate(110px) rotate(-330deg);
  background: linear-gradient(to bottom right, #8EA3D5 0%, #9D8FD4 100%)
}

.circle .note:disabled {
  background: gainsboro;
}