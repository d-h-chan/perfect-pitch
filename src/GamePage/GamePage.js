import React, { Component } from 'react';
import AnswerButtons from '../AnswerButtons/AnswerButtons.js';
import Context from '../ContextManagement/Context.js'
import { idToFrequencyMap, frequencyToNoteMap, noteToFrequencyMap } from '../store.js';

class GamePage extends Component {

  static contextType = Context;

  constructor(props) {
    super(props);
    this.audioContext = new AudioContext();
  }

  playNote = (frequency) => {
    let oscillator = this.audioContext.createOscillator();
    oscillator.connect(this.audioContext.destination);
    oscillator.frequency.setValueAtTime(frequency, this.audioContext.currentTime); // value in hertz
    oscillator.start(this.audioContext.currentTime)
    oscillator.stop(this.audioContext.currentTime + 1.5)
  }

  handleClick = () => {
    //let num = Math.floor(Math.random() * 11) + 1;
    //let frequency = idToFrequencyMap[num]
    //this.context.setCurrentNote(num);
    //console.log(frequency)

    //this.playNote(frequency);
    this.playNote(this.context.currentNote)
  }

  generateRandomNote = () => {
    const num = Math.floor(Math.random() * 11) + 1;
    const frequency = idToFrequencyMap[num]
    console.log(frequency)
    return frequency
  }

  render() {
    return (
      <>
        <h1>Game Screen</h1>
        <button onClick={this.handleClick}>Play Sound</button>
        <br></br>
        <AnswerButtons></AnswerButtons>
        <br></br>
        <Context.Consumer>
          {({ currentNote, setCurrentNote }) => (
            <button onClick={() => setCurrentNote(this.generateRandomNote())}>
              Switch Note (Current: {currentNote})
          </button>
          )}
        </Context.Consumer>

      </>
    );
  }
}

export default GamePage;