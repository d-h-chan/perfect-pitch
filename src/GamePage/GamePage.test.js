import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import GamePage from './GamePage'

it('renders without errors', () => {
  const wrapper = shallow(<GamePage />)
  expect(toJson(wrapper)).toMatchSnapshot()
})