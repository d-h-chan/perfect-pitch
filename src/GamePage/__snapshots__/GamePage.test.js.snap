// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a .Note by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`renders without errors 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
